### PR TITLE
Node 6 ES 6 and Airbnb styleguide

### PR DESCRIPTION
**Pull request by julianbei**
enforced the coding style guide from:
https://github.com/airbnb/javascript

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ A client for the HTTP API of HashiCorp's [Vault] written for Node.js.
 
 
 ## install
+make sure to use node.js version > 6
```
Comment (kr1sp1n) [inline README.md]: >=

File: example/audits.js [hunk 2]
```diff
@@ -2,13 +2,14 @@
 
 process.env.DEBUG = 'node-vault'; // switch on debug mode
 const vault = require('./../src/index')();
-const file_path = '/tmp/logs/test.log';
+const path = '/tmp/logs/test.log';
 
 vault.audits()
 .then((result) => {
   if (!result.hasOwnProperty('testlog/')) {
```
Comment (kr1sp1n) [inline example/audits.js]: guard clause

File: example/auth_github.js [hunk 3]
```diff
@@ -5,20 +5,22 @@ const vault = require('./../src/index')();
...
 .then((result) => {
   if (!result.hasOwnProperty('github/')) {
-    return vault.enableAuth({ mount_point: mountPoint, type: 'github', description: 'GitHub auth' });
+    return vault.enableAuth({
+      mount_point: mountPoint,
+      type: 'github',
+      description: 'GitHub auth',
+    });
   }
+  return undefined;
```
Comment (kr1sp1n) [inline example/auth_github.js]: guard clause

File: example/auth_github.js [hunk 4]
```diff
@@ -5,20 +5,22 @@ const vault = require('./../src/index')();
...
-  return vault.write('auth/github/config', { organization: org })
-  .then(() => vault.write('auth/github/map/teams/' + team, { value: 'root' }))
-  .then(() => vault.githubLogin({ token: token }))
-  .then(console.log);
-})
+.then(() => vault.write('auth/github/config', { organization: org }))
+.then(() => vault.write(`auth/github/map/teams/${team}`, { value: 'root' }))
+.then(() => vault.githubLogin({ token }))
+.then(console.log)
 .catch((err) => console.error(err));
```
Comment (kr1sp1n) [inline example/auth_github.js]: fix this

File: example/auth_userpass.js [hunk 5]
```diff
@@ -10,10 +10,15 @@ const password = 'foo';
...
 .then((result) => {
   if (!result.hasOwnProperty('userpass/')) {
-    return vault.enableAuth({ mount_point: mountPoint, type: 'userpass', description: 'userpass auth' });
+    return vault.enableAuth({
+      mount_point: mountPoint,
+      type: 'userpass',
+      description: 'userpass auth',
+    });
   }
+  return undefined;
```
Comment (kr1sp1n) [inline example/auth_userpass.js]: guard clause

File: example/mount_postgresql.js [hunk 6]
```diff
@@ -16,20 +16,20 @@ const configure = () => {
...
-    .then(run);
-  } else {
-    return run();
+    return vault.mount({
+      mount_point: 'postgresql',
+      type: 'postgresql',
+      description: 'postgresql mount test',
+    }).then(run);
   }
+  return run();
```
Comment (kr1sp1n) [inline example/mount_postgresql.js]: guard clause

File: test/unit.js [hunk 7]
```diff
@@ -231,79 +216,69 @@ describe('node-vault', function () {
...
           const fn = vault[name];
           const promise = fn();
-          request.calledOnce.should.be.ok;
-          promise.should.be.Promise();
-          promise.then(function (body) {
-            return done();
-          })
-          .catch(fail(done));
+          request.calledOnce.should.be.ok();
+          promise.should.be.a.Promise;
```
Comment (kr1sp1n) [inline test/unit.js]: find fix

File: test/unit.js [hunk 8]
```diff
@@ -231,79 +216,69 @@ describe('node-vault', function () {
...
+        it('should handle config with schema property', done => {
           const name = 'myGeneratedFunction';
           vault.generateFunction(name, configWithSchema);
           const fn = vault[name];
           const promise = fn({ testProperty: 3 });
-          promise.then(function (body) {
-            return done();
-          })
-          .catch(fail(done));
+          promise.then(() => done())
```
Comment (kr1sp1n) [inline test/unit.js]: fix -> then(done)

File: test/unit.js [hunk 9]
```diff
@@ -231,79 +216,69 @@ describe('node-vault', function () {
...
-    describe('request(options)', function () {
-
-      it('should handle undefined options', function (done) {
+    describe('request(options)', () => {
+      it('should handle undefined options', done => {
         const promise = vault.request();
-        promise.catch(function (err) {
-          return done();
-        });
+        promise.catch(() => done());
```
Comment (kr1sp1n) [inline test/unit.js]: fix this test to be meaningful

Comment (kr1sp1n) [inline test/unit.js]: .then(()=> done(new Error('should not be called'))

**Comment by julianbei** [comment]
there is an error with the build pipeline:
```

Attaching to nodevault_test_1
test_1 | 
test_1 | > node-vault@0.5.1 test /app
test_1 | > _mocha --recursive
test_1 | 
test_1 | module.js:442
test_1 |     throw err;
test_1 |     ^
test_1 | 
test_1 | Error: Cannot find module 'should'
test_1 |     at Function.Module._resolveFilename (module.js:440:15)
test_1 |     at Function.Module._load (module.js:388:25)
test_1 |     at Module.require (module.js:468:17)
test_1 |     at require (internal/module.js:20:19)
test_1 |     at Object.<anonymous> (/app/test/integration.js:1:63)
test_1 |     at Module._compile (module.js:541:32)
test_1 |     at Object.Module._extensions..js (module.js:550:10)
test_1 |     at Module.load (module.js:458:32)
test_1 |     at tryModuleLoad (module.js:417:12)
test_1 |     at Function.Module._load (module.js:409:3)
test_1 |     at Module.require (module.js:468:17)
test_1 |     at require (internal/module.js:20:19)
test_1 |     at /app/node_modules/mocha/lib/mocha.js:219:27
test_1 |     at Array.forEach (native)
test_1 |     at Mocha.loadFiles (/app/node_modules/mocha/lib/mocha.js:216:14)
test_1 |     at Mocha.run (/app/node_modules/mocha/lib/mocha.js:468:10)
test_1 |     at Object.<anonymous> (/app/node_modules/mocha/bin/_mocha:403:18)
test_1 |     at Module._compile (module.js:541:32)
test_1 |     at Object.Module._extensions..js (module.js:550:10)
test_1 |     at Module.load (module.js:458:32)
test_1 |     at tryModuleLoad (module.js:417:12)
test_1 |     at Function.Module._load (module.js:409:3)
test_1 |     at Module.runMain (module.js:575:10)
test_1 |     at run (bootstrap_node.js:352:7)
test_1 |     at startup (bootstrap_node.js:144:9)
test_1 |     at bootstrap_node.js:467:3
test_1 | npm ERR! Test failed.  See above for more details.
nodevault_test_1 exited with code 1
The command "docker-compose up --force-recreate test" exited with 0.
```

The pipeline does not fail if a test fails.

**Comment by julianbei** [comment]
actually the tests are working with docker-compose:

```
julianbei% docker-compose up --force-recreate test
WARNING: The COVERALLS_REPO_TOKEN variable is not set. Defaulting to a blank string.
Recreating nodevault_postgres_1
Recreating nodevault_vault_1
Recreating nodevault_test_1
Attaching to nodevault_test_1
test_1 | 
test_1 | > node-vault@0.5.1 test /app
test_1 | > _mocha --recursive
test_1 | 
test_1 | 
test_1 | 
test_1 |   integration
test_1 |     node-vault
      ✓ should initialize a vault server (46ms)
      ✓ should show the current status of the vault server
test_1 | 
test_1 |   node-vault
test_1 |     module
      ✓ should export a function that returns a new client
test_1 |     client
test_1 |       help(path, options)
        ✓ should response help text for any path
        ✓ should handle undefined options
test_1 |       write(path, data, options)
        ✓ should write data to path
        ✓ should handle undefined options
test_1 |       read(path, options)
        ✓ should read data from path
        ✓ should handle undefined options
test_1 |       delete(path, options)
        ✓ should delete data from path
        ✓ should handle undefined options
test_1 |       handleVaultResponse(response)
        ✓ should return a function that handles errors from vault server
        ✓ should return a Promise with the body if successful
        ✓ should return a Promise with the error if failed
test_1 |       generateFunction(name, config)
        ✓ should generate a function with name as defined in config
test_1 |         generated function
          ✓ should return a promise
          ✓ should handle config with schema property
          ✓ should handle invalid arguments via schema property
test_1 |       request(options)
        ✓ should reject if options are undefined
        ✓ should handle undefined path in options
        ✓ should handle undefined method in options
test_1 | 
test_1 | 
test_1 |   21 passing (161ms)
test_1 | 
nodevault_test_1 exited with code 0
```

**Comment by julianbei** [comment]
Build is slow or down because of registry.npm.org having problems with reads.
```
Investigating
We've confirmed that the situation is an on-going, albeit intermittent problem and are investigating.
Posted 5 minutes ago. Jul 25, 2016 - 14:05 UTC
Monitoring
Starting at 12:30 UTC we saw an increase in 503 responses from the registry. The situation seemed to have resolved at 13:10 UTC, but we will continue monitoring and investigating the reasons of this outage.
Posted about 1 hour ago. Jul 25, 2016 - 13:19 UTC
```

**Comment by julianbei** [comment]
the previously failing tests are running through now.. 💯  👍  💚

**Comment by julianbei** [comment]
if the build script is running as it should, we should fix the package.json to use the following gulp tasks:

### test (including linting)
``` gulp test ```
### build (including linting and testing)
``` gulp build ```
### watch (continuously executing tests and linting when a file changes)
``` gulp watch ```

**Comment by julianbei** [comment]
- deleted the script folder because of having the feature.md generator in the gulp script the folder no langer has a `bleiberecht`
- `npm docs` which is an alias for `gulp docs` will now generate the documentation with docco and the feature.md docs
- the documentation will be generated in the docs subfolder

please review and discuss enhancements

**Pull request merged**